scripts/prepare_data.py

```diff
@@ -62,67 +164,120 @@ def add_tile_id(row):
+            fp_labels_gdf['year'] = fp_labels_gdf.year.astype(int)
+            fp_labels_4326_gdf = fp_labels_gdf.to_crs(epsg=4326).drop_duplicates(subset=['geometry', 'year'])
+        else:
+            fp_labels_4326_gdf = fp_labels_gdf.to_crs(epsg=4326).drop_duplicates(subset=['geometry'])
+        fp_labels_4326_gdf['CATEGORY'] = 'mineral extraction site'
+        fp_labels_4326_gdf['SUPERCATEGORY'] =  'land usage'
+
+        nb_fp_labels = len(fp_labels_gdf)
+        logger.info(f"There are {nb_fp_labels} polygons in {FP_SHPFILE}")
+
+        filename = 'FP.geojson'
+        filepath = os.path.join(OUTPUT_DIR, filename)
+        fp_labels_4326_gdf.to_file(filepath, driver='GeoJSON')
+        written_files.append(filepath)  
+        logger.success(f"{DONE_MSG} A file was written: {filepath}")
+        labels_4326_gdf = pd.concat([labels_4326_gdf, fp_labels_4326_gdf], ignore_index=True)
+
+    # Tiling of the AoI
+    logger.info("- Get the label boundaries")  
+    boundaries_df = labels_4326_gdf.bounds
+    logger.info("- Tiling of the AoI")  
+    tiles_4326_aoi_gdf = aoi_tiling(boundaries_df)
+    tiles_4326_labels_gdf = gpd.sjoin(tiles_4326_aoi_gdf, labels_4326_gdf, how='inner', predicate='intersects')
+
+    # Tiling of the AoI from which empty tiles will be selected
+    if EPT_SHPFILE:
+        EPT_aoi_gdf = gpd.read_file(EPT_SHPFILE)
+        EPT_aoi_4326_gdf = EPT_aoi_gdf.to_crs(epsg=4326)
+        assert_year(labels_4326_gdf, EPT_aoi_4326_gdf, 'empty_tiles', EPT_YEAR)
+        
+        if EPT == 'aoi':
+            logger.info("- Get AoI boundaries")  
+            EPT_aoi_boundaries_df = EPT_aoi_4326_gdf.bounds
+
+            # Get tile coordinates and shapes
+            logger.info("- Tiling of the empty tiles AoI")  
+            empty_tiles_4326_all_gdf = aoi_tiling(EPT_aoi_boundaries_df)
+            # Delete tiles outside of the AoI limits 
+            empty_tiles_4326_aoi_gdf = gpd.sjoin(empty_tiles_4326_all_gdf, EPT_aoi_4326_gdf, how='inner', lsuffix='ept_tiles', rsuffix='ept_aoi')
+            # Attribute a year to empty tiles if necessary
+            if 'year' in labels_4326_gdf.keys():
+                if isinstance(EPT_YEAR, int):
+                    empty_tiles_4326_aoi_gdf['year'] = int(EPT_YEAR)
+                else:
+                    empty_tiles_4326_aoi_gdf['year'] = np.random.randint(low=EPT_YEAR[0], high=EPT_YEAR[1], size=(len(empty_tiles_4326_aoi_gdf)))
+            elif EPT_SHPFILE and EPT_YEAR: 
+                logger.warning("No year column in the label shapefile. The provided empty tile year will be ignored.")
+        elif EPT == 'shp':
+            if EPT_YEAR:
+                logger.warning("A shapefile of selected empty tiles are provided. The year set for the empty tiles in the configuration file will be ignored")
+                EPT_YEAR = None
+            empty_tiles_4326_aoi_gdf = EPT_aoi_4326_gdf.copy()
+
+    # Get all the tiles in one gdf 
+    if EPT_SHPFILE:
```

Comment:
This line can be deleted and the following one will be considered in the `if` of line 218.